Weatherman.py

```diff
@@ -0,0 +1,277 @@
+    for iterator in range(0, len(minimum_temperature_values)-1):
+        if maximum_temperature_values[iterator] == '':
+            print('', end='')
+        else:
+            print(iterator+1, end='')
+            display_colored_barchart(float(maximum_temperature_values[iterator]), Red)
+            display_colored_barchart(float(minimum_temperature_values[iterator]), Blue)
+            print(White, f'{int(maximum_temperature_values[iterator])}C-'
+                  f'{int(minimum_temperature_values[iterator])}C', end='')
+            print('')
+
+
+def display_colored_barchart(number, color):
+    number_of_stars = int(number) + 1
+    for iterator in range(1, number_of_stars):
+        print(color + '+', end='')
+
+
+# For year
+
+def make_filename(year, number):
+    month_number = str(number)
+    datetime_object = datetime.strptime(month_number, "%m")
+    month_name = datetime_object.strftime("%b")
+    file_name = f'Murree_weather_{year}_{month_name}.txt'
+    return file_name
+
+
+def run_weather_man_year(year):
+    maximum_temperatures = []
+    minimum_temperatures = []
+    minimum_temperature_dates = []
+    maximum_temperature_dates = []
+    mean_humidity = []
+    mean_humidity_dates = []                 # I need these variables in and after the for loop
+    for iterator in range(1, 13):
+        file_name = make_filename(year, iterator)
+        if path.exists(file_name):
+            max_temperature_values, min_temperature_values, mean_humidity_values, date_reading = \
+                    make_temperature_values(file_name)     # 1st function
+        # we are calculating max month temp of every month in every iteration
+            index_max = calculate_maximum_temperature(max_temperature_values)
+            maximum_temperature = max_temperature_values[index_max]
+            maximum_temperatures.append(maximum_temperature)
+            maximum_temperature_dates.append(calculate_date(date_reading, index_max))
+        # minimum
+            index_min = calculate_minimum_temperature(min_temperature_values)
+            minimum_temperature = min_temperature_values[index_min]
+            minimum_temperatures.append(minimum_temperature)
+            minimum_temperature_dates.append(calculate_date(date_reading, index_min))
+        # humidity
+            index_mean = calculate_maximum_temperature(mean_humidity_values)
+            mean_humidity1 = mean_humidity_values[index_max]
+            mean_humidity.append(mean_humidity1)
+            mean_humidity_dates.append(calculate_date(date_reading, index_mean))
+    calculate_results(maximum_temperatures, maximum_temperature_dates,
+                      minimum_temperatures, minimum_temperature_dates,
+                      mean_humidity, mean_humidity_dates)             # 2nd function
+
+# 1st function
+
+
+def calculate_date(date_values, index):
+    return date_values[index]
+
+
+def make_temperature_values(file_name):
```

Comment:
Are you making temperature values or reading them? Think twice before naming a variable.

Reply:
making a list of temperature values while reading from a file